File: backend/app/schemas/compliance.py
from pydantic import BaseModel, Field


class ComplianceScore(BaseModel):
    score: int = Field(ge=0, le=100)
    rating: str


class RegulatoryAlert(BaseModel):
    id: str
    title: str
    summary: str
    severity: str
    source_url: str
    published_at: str


class ComplianceDashboard(BaseModel):
    score: ComplianceScore
    active_alerts: list[RegulatoryAlert]

File: shared/contracts/compliance.ts
export type ComplianceScore = {
  score: number;
  rating: string;
};

export type RegulatoryAlert = {
  id: string;
  title: string;
  summary: string;
  severity: string;
  source_url: string;
  published_at: string;
};

export type ComplianceDashboard = {
  score: ComplianceScore;
  active_alerts: RegulatoryAlert[];
};
